src/app/core/ui/navigation/navigation/navigation.component.ts

```diff
@@ -82,16 +88,16 @@ export class NavigationComponent {
    */
   private computeActiveLink(newUrl: string): string {
     // conservative filter matching all items that could fit to the given url
-    const items: MenuItem[] = this.menuItems.filter((item) =>
-      newUrl.startsWith(item.link),
+    const items: MenuItem[] = this.menuItems.filter(
+      (item) => typeof item.link === "string" && newUrl.startsWith(item.link),
     );
     switch (items.length) {
       case 0:
         return "";
       case 1:
-        const link = items[0].link;
+        const target = items[0].link as string;
```

Comment:
@sadaf895 , I think we don't need to rename this anymore. Do you want to undo these changes and explicit types ("as string") to keep the changes to a minimum?